src/main/kotlin/net/atos/zac/app/signaleringen/SignaleringenRestService.kt

```diff
@@ -0,0 +1,171 @@
+import net.atos.client.zgw.zrc.model.Zaak
+import net.atos.zac.app.informatieobjecten.converter.RESTInformatieobjectConverter
+import net.atos.zac.app.informatieobjecten.model.RESTEnkelvoudigInformatieobject
+import net.atos.zac.app.signaleringen.converter.RESTSignaleringInstellingenConverter
+import net.atos.zac.app.signaleringen.model.RESTSignaleringInstellingen
+import net.atos.zac.app.taken.converter.RESTTaakConverter
+import net.atos.zac.app.taken.model.RESTTaak
+import net.atos.zac.app.zaken.converter.RESTZaakOverzichtConverter
+import net.atos.zac.app.zaken.model.RESTZaakOverzicht
+import net.atos.zac.authentication.LoggedInUser
+import net.atos.zac.flowable.TakenService
+import net.atos.zac.identity.IdentityService
+import net.atos.zac.signalering.SignaleringenService
+import net.atos.zac.signalering.model.Signalering
+import net.atos.zac.signalering.model.SignaleringInstellingen
+import net.atos.zac.signalering.model.SignaleringInstellingenZoekParameters
+import net.atos.zac.signalering.model.SignaleringSubject
+import net.atos.zac.signalering.model.SignaleringType
+import net.atos.zac.signalering.model.SignaleringZoekParameters
+import nl.lifely.zac.util.NoArgConstructor
+import org.flowable.task.api.TaskInfo
+import java.time.ZonedDateTime
+import java.util.UUID
+
+@Path("signaleringen")
+@Consumes(MediaType.APPLICATION_JSON)
+@Produces(MediaType.APPLICATION_JSON)
+@Singleton
+@NoArgConstructor
+@Suppress("LongParameterList")
+class SignaleringenRestService @Inject constructor(
+    private val signaleringenService: SignaleringenService,
+    private val zrcClientService: ZRCClientService,
+    private val takenService: TakenService,
+    private val drcClientService: DRCClientService,
+    private val identityService: IdentityService,
+    private val restZaakOverzichtConverter: RESTZaakOverzichtConverter,
+    private val restTaakConverter: RESTTaakConverter,
+    private val restInformatieobjectConverter: RESTInformatieobjectConverter,
+    private val restSignaleringInstellingenConverter: RESTSignaleringInstellingenConverter,
+    private val loggedInUserInstance: Instance<LoggedInUser>,
+) {
+    @GET
+    @Path("/latest")
+    fun latestSignaleringen(): ZonedDateTime? {
+        val parameters = SignaleringZoekParameters(loggedInUserInstance.get())
+        return signaleringenService.latestSignalering(parameters)
+    }
+
+    @GET
+    @Path("/zaken/{type}")
+    fun listZakenSignaleringen(
+        @PathParam("type") signaleringsType: SignaleringType.Type
+    ): List<RESTZaakOverzicht> {
+        val parameters = SignaleringZoekParameters(loggedInUserInstance.get())
+            .types(signaleringsType)
+            .subjecttype(SignaleringSubject.ZAAK)
+        return signaleringenService.listSignaleringen(parameters).stream()
+            .map { signalering: Signalering -> zrcClientService.readZaak(UUID.fromString(signalering.subject)) }
+            .map { zaak: Zaak? -> restZaakOverzichtConverter.convert(zaak!!) }
```

Comment:
I think this code needs to be tweaked a bit. `zaak` is treated as nullable here but if you look at the actual zrcClientService you will see that it never returns null (if will throw an exception if it fails to find the zaak). So this line should be: 
```suggestion
            .map { restZaakOverzichtConverter.convert(it) }
```

>@return {@link Zaak}. Never NULL!